src/ast_editor_state.mli

```diff
@@ -0,0 +1,39 @@
+open Import
+
+type t
+
+val make : unit -> t
+
+val find_original_doc_by_pp_uri :
+  uri_string:string -> ('a, string, 'b) Import.Map.t -> 'a option
+
+val find_webview_by_doc : t -> document:TextDocument.t -> WebView.t option
+
+val set_changes_tracking : t -> TextDocument.t -> TextDocument.t -> unit
+
+val get_original_mode : t -> bool
+
+val set_original_mode : t -> bool -> unit
+
+val get_hover_disposable : t -> Disposable.t option
+
+val set_hover_disposable : t -> Disposable.t option -> unit
+
+val set_origin_changed : t -> data:bool -> key:string -> unit
+
+val entry_exists : t -> string -> string -> bool
```

Comment:
The arguments for this function need labels.

Reply:
Done in 001f9964951a21cccebf25092f417f31be9f28a7